TLM/TLM/Manager/Impl/RoutingManager.cs

```diff
@@ -480,9 +480,71 @@ public class RoutingManager
                 nodeIsSplitJunction = numOutgoing > 1;
             }
 
-            // bool isNextRealJunction = prevSegGeo.CountOtherSegments(startNode) > 1;
+            bool applyHighwayMergingRules = false;
+            bool nodeHasConnections = LaneConnectionManager.Instance.HasNodeConnections(nodeId);
+            if (extendedLogRouting) {
+                Log._Debug($"Options.highwayMergingRules={Options.highwayMergingRules}, nodeHasTrafficLights={nodeHasTrafficLights} nodeHasPrioritySigns={nodeHasPrioritySigns}, nodeHasConnections={nodeHasConnections}");
+            }
+            if (Options.highwayMergingRules && !nodeHasTrafficLights && !nodeHasPrioritySigns && !nodeHasConnections) {
+                // determine if junction is a simple junction (highway rules only apply to simple junctions)
+                int numIncomingSegents = 0;
+                int numOutgoingSegments = 0;
+                bool laneSwitching = false;
+                bool highwayOnly = true;
+
+                for (int segIndex = 0; segIndex < 8; ++segIndex) {
+                    ushort segmentId = netNode.GetSegment(segIndex);
+                    if (segmentId == 0) {
+                        continue;
+                    }
+
+                    ref NetSegment netSegment = ref segmentId.ToSegment();
+                    bool startNode = netSegment.IsStartNode(nodeId);
+                    highwayOnly &= netSegment.Info?.m_netAI is RoadBaseAI ai && ai.m_highwayRules; 
+
+                    ExtSegmentEnd segEnd = segEndMan.ExtSegmentEnds[segEndMan.GetIndex(segmentId, startNode)];
+                    if (segEnd.incoming) {
+                        ++numIncomingSegents;
+                        laneSwitching |= JunctionRestrictionsManager.Instance.IsLaneChangingAllowedWhenGoingStraight(segmentId, startNode);
+                    }
+
+                    if (segEnd.outgoing) {
+                        ++numOutgoingSegments;
+                    }
+                }
+
+                if (numOutgoingSegments == 1 && numIncomingSegents == 2 && !laneSwitching && highwayOnly) {
```

Comment:
Hmm, so it's no longer restricted to one-way roads like legacy (line 552-553 below)?

Reply:
That is actually a very interesting point. What should we do in case of 2 way highways like [the highway over wall collection](https://steamcommunity.com/workshop/filedetails/?id=1336987320)?

Reply:
practically `numOutgoingSegments == 1 && numIncomingSegents == 2` only works for oneway (unless if you really try to break it but even then lane arithmetic won't add up).

I will add condition for oneway to clarify/full-proof the code.

Reply:
BTW I just noticed when using legacy highway rule option one lane may not be not used at all 
![image](https://user-images.githubusercontent.com/26344691/167289396-15a87968-7fe9-4e67-a19b-632596778c13.png)

Reply:
OK now I only check for oneway highways.

Reply:
I created #1561